### ISSUE TITLE
Issue in saving the checkpoint after epoch

### ISSUE THREAD

**Issue by rkkumar16** [NVIDIA/NeMo#2738]
**Issue in saving the checkpoint after epoch**
After the epoch runs successfully it tries to save the model checkpoint file in storage and throws exception

**MisconfigurationException: ModelCheckpoint(monitor='val_loss') not found in the returned metrics: ['lr', 'train_loss']**

**Steps/Code to reproduce bug**
Following the IMDB dataset structures put up custom dataset for classification model training and run the training epochs. While it tries to save the checkpoint exception is generated.

**Expected behavior**
Model should be saved to the storage after each epoch or number of epochs and next iteration should be running.

[Screenshot from 2021-08-27 12-57-25](https://user-images.githubusercontent.com/41611198/131109161-0252e098-3d4c-46d1-be31-b62a61d00e90.png)

**Environment overview (please complete the following information)**

 - Environment location: Kubernetes pod
 - Method of NeMo install: NeMo image as base

**Additional context**
NeMo version 1.0.1 and Python version 3.8
Model used is bert base uncased

**Comment by rkkumar16** [comment]
I have followed the IMDB dataset structure and it had test and train folders only. So, there is no validation folder. Number of epochs I have set are 20

**Comment by VahidooX** [comment]
It looks like that the issue is that checkpoints are saved based on the validation loss or accuracy and as you do not have any validation set, no checkpoint is saved. But it should ignore this when no validation is set. I would take a look into that. In the meantime, please add the test set or a small part of train as dev set so that it stores the checkpoints.
